std/utf.d

```diff
@@ -1005,8 +1005,7 @@ alias UseReplacementDchar = Flag!"useReplacementDchar";
         sequence and useReplacementDchar is UseReplacementDchar.no
   +/
 dchar decode(UseReplacementDchar useReplacementDchar = UseReplacementDchar.no, S)(auto ref S str, ref size_t index)
-    if (!isSomeString!S &&
-        isRandomAccessRange!S && hasSlicing!S && hasLength!S && isSomeChar!(ElementType!S))
+    if (isSomeString!S || isRandomAccessRange!S && hasSlicing!S && hasLength!S && isSomeChar!(ElementType!S))
```

Comment:
I combined the range overload with the `if (isSomeString!S)` one directly below.

The only reason they were separate previously, was to make the `isSomeString` version `@trusted`. But my other changes make that unnecessary now.